### PR TITLE
feat(SidebarSearchModal): support customizing content

### PR DESCRIPTION
**Pull request by kuangp**
Signed-off-by: Phil Kuang <pkuang@factset.com>

## Hey, I just made a Pull Request!

Add support for rendering a custom `SidebarSearchModal`. Also updated the example app with an example custom dialog shown below:
Screen Shot 2022-03-23 at 3 03 16 PM

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/app/src/components/Root/Root.tsx [hunk 1]
```diff
@@ -88,7 +89,9 @@ export const Root = ({ children }: PropsWithChildren<{}>) => (
     <Sidebar>
       <SidebarLogo />
       <SidebarGroup label="Search" icon={<SearchIcon />} to="/search">
-        <SidebarSearchModal />
+        <SidebarSearchModal>
```
Comment (iamEAP) [inline packages/app/src/components/Root/Root.tsx:92]: Getting this in the example app is nice!  I wonder if it'd be good to document elsewhere too.  Perhaps storybook?  The sidebar story could use a little love, perhaps.

**Review by iamEAP — COMMENTED** [review]
Thanks @kuangp!  You've been killin' it these past couple weeks.  I've got one minor comment about the interface for the child function, but once that's addressed, I'm good with shipping this.

**Overall** A big thing I want for us to try and tackle before we put a nice, stable 1.0 bow on the Search plugin is improved composability.  Right now, things are very black and white: either you use what's provided, or you build your own thing from scratch.

I like this PR as a step in the right direction for the modal (you get utilities around the surface itself, but total control over the experience).  ...But I'm hoping we can have a tidier option in the future for folks who basically like the layout, but just want to use custom result types, specific filters, etc.

**Comment by kuangp** [comment]
> But I'm hoping we can have a tidier option in the future for folks who basically like the layout, but just want to use custom result types, specific filters, etc.

That would be awesome! Would definitely make it much easier to customize without re-implementing the entire page!

**Comment by kuangp** [comment]
@iamEAP update according to your suggestions! let me know if there's anything else you think we need here!

**Review by iamEAP — APPROVED** [review]
Sweet!  Thanks again, @kuangp.

**Comment by emmaindal** [comment]
@backstage/reviewers can we get your eyes on this one. Thank you!

**Review by jhaals — APPROVED** [review]
Looking good! 👍

**Pull request merged**